base/ntuple.jl

```diff
@@ -14,8 +14,9 @@ julia> ntuple(i -> 2*i, 4)
 (2, 4, 6, 8)
 ```
 """
-@inline function ntuple(f::F, n::Integer) where F
-    # marked inline since this benefits from constant propagation of `n`
+@inline function ntuple(f::F, m::Integer) where F
+    # marked inline since this benefits from constant propagation of `m`
+    n = Int(m)::Int
```

Comment:
This can be written as
```suggestion
    n::Int = m
```
which is syntax sugar for `n = convert(Int, m)::Int

Reply:
`convert(Int, m)` is not the same as `Int(m)` though

Reply:
It seems fine though, no?
https://github.com/JuliaLang/julia/blob/17445fe752b7b99633ca306af0981baca9f66bda/base/number.jl#L6-L7